core-lib/TestSuite/LiteralObjectTests.som

```diff
@@ -0,0 +1,102 @@
+(* Copyright (c) 2001-2015 see AUTHORS file
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the 'Software'), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*)
+class ObjectTests usingPlatform: platform testFramework: minitest = (
+| private TestContext = minitest TestContext. |
+) (
+  public class ObjectTest = TestContext ()(
+
+    public testEquality = (
+      | a b |
+      a := objL()().
+      b := objL()().
+      self deny: a == b.
+    )
+  
+    public testMethodLocalRead = (
+      | x a |
+      x := 1.
+      a := objL()(public b = (^ x)).
+      self assert: a b = 1.
```

Comment:
I would prefer the use of `assert: a b equals: 1` makes for nicer error messages.
and I think, I'd drop the `self` since it isn't strictly necessary.